src/lib.rs

```diff
@@ -411,7 +415,7 @@ impl<S: SuffixScheme> FileRotate<S> {
         path: P,
         suffix_scheme: S,
         content_limit: ContentLimit,
-        compression: Compression,
+        #[cfg(feature = "compression")] compression: Compression,
         #[cfg(unix)] mode: Option<u32>,
```

Comment:
Unrelated to this commit, but this code forces cross-platform code that uses this library to specify `#[cfg(unix)]`. I do not think this design is good. @Ploppz